### DISCUSSION TITLE
Linking XAML Page on Tile

### DISCUSSION THREAD

**Discussion by Nopsii** [MahApps/MahApps.Metro#4173] [Q&A] [answered]
Im currently working on a Progra where the frontpage is are Tiles in a Wrappanel. In an older Projekt i did it with an Tabcontrol where you can add an MetroTabitem and just say <local:UserControl/> and linked it to the tab. 
How can i do that with a Tile?
Is there an similar way or do i have to make an click event and make pages visible or hidden?

Thank you for any help you can provide.

**Comment by timunie** [comment]
Hi @Nopsii 

using the `Click-Event` or an `ICommand` is what comes me in mind. And then change the `DataContext` and maybe use a `DataTemplateSelector`. But as we don't know your code it is hatd to guess. There are thousands of possibilities.

Happy coding
Tim

**Reply by Nopsii** [reply]
Thanks For your answer. Sorry i forgot to post my code too.
<Grid Background="LightCyan">

        <WrapPanel Orientation="Horizontal" ItemHeight="200" ItemWidth="392" Margin="222,175,194,165">
            <Controls:Tile Title="Shop" Controls:ControlsHelper.MouseOverBorderBrush="{DynamicResource MahApps.Brushes.ThemeForeground}" Background="Teal" HorizontalTitleAlignment="Right" Height="Auto" Width="Auto">
                <iconPacks:PackIconBootstrapIcons Kind="Cart2" Height="150" Width="200"/>
            </Controls:Tile>
            <Controls:Tile Title="Data" Controls:ControlsHelper.MouseOverBorderBrush="{DynamicResource MahApps.Brushes.ThemeForeground}" Background="Teal" HorizontalTitleAlignment="Right" Height="Auto" Width="Auto">
                <iconPacks:PackIconBootstrapIcons Kind="Cursor" Height="150" Width="200"/>
            </Controls:Tile>
            <Controls:Tile Title="Settings" Controls:ControlsHelper.MouseOverBorderBrush="{DynamicResource MahApps.Brushes.ThemeForeground}" Background="Teal" HorizontalTitleAlignment="Right" Height="Auto" Width="Auto">
                <iconPacks:PackIconBootstrapIcons Kind="Archive" Height="150" Width="200"/>
             </Controls:Tile>
            <Controls:Tile Title="Login" Controls:ControlsHelper.MouseOverBorderBrush="{DynamicResource MahApps.Brushes.ThemeForeground}" Background="Teal" HorizontalTitleAlignment="Right" Height="Auto" Width="Auto">
                <iconPacks:PackIconBootstrapIcons Kind="CardList" Height="150" Width="200"/>
            </Controls:Tile>
            <Controls:Tile Title="Test1" Controls:ControlsHelper.MouseOverBorderBrush="{DynamicResource MahApps.Brushes.ThemeForeground}" Background="Teal" HorizontalTitleAlignment="Right" Height="Auto" Width="Auto">
                <iconPacks:PackIconBootstrapIcons Kind="Printer" Height="150" Width="200"/>
            </Controls:Tile>
            <Controls:Tile Title="Test2" Controls:ControlsHelper.MouseOverBorderBrush="{DynamicResource MahApps.Brushes.ThemeForeground}" Background="Teal" HorizontalTitleAlignment="Right" Height="Auto" Width="Auto">
                <iconPacks:PackIconBootstrapIcons Kind="BarChart" Height="150" Width="200"/>
            </Controls:Tile>
        </WrapPanel>


That is the XAML for the first page and when i click on one of them i want it to change to the linked XAML(UserControl) like when you use a TabControl

**Reply by timunie** [reply]
Hi @Nopsii 

this is only the view. Having a sample App would be much better. Right now I would use the `Click`-Event.

Happy coding
Tim

**Comment by Nopsii** [comment, marked as answer]
Hi i found a solution to this. I was a little bit busy the last days but wanted to share my code (maybe it helps someone).
Sadly i found no option to link it directly in the XAML where i put the code of the `Tile` like using a `TabControl` and put `local:myXAML` as content.

I made a public function in my MainWindow and a `Frame` in the XAML (or wherever you want to navigate from).

```
public static void callPage(string _myPage)
        {
            var test = new Uri(_myPage + ".xaml", UriKind.RelativeOrAbsolute);
            mainFrame.Source = null;
            mainFrame.Source = new Uri(_myPage + ".xaml", UriKind.RelativeOrAbsolute);
        }
```
```
 <Frame x:Name="MainFrame" NavigationUIVisibility="Hidden"/>
```

 Then i made a click event on my `Tile` and i just call my function and put the XAML name as statement.

```
callPage("myXAML");
```

Thanks for helping and maybe there will be added a way to put a XAML as content directly in the `Tile`

Nopsii
